apps/web/src/presentation/common/presenter/app/AppPresenter.ts

```diff
@@ -1,19 +1,59 @@
 import { CreateTransactionService } from '../../../../application/services/CreateTransaction/CreateTransactionService'
+import { GetSelectedNetworkService } from '../../../../application/services/GetSelectedNetwork/GetSelectedNetworkService'
+import { ListNetworksService } from '../../../../application/services/ListNetworks/ListNetworksService'
+import { SwitchNetworkService } from '../../../../application/services/SwitchNetwork/SwitchNetworkService'
 import { Presenter } from '../../base/Presenter'
 import { AppPresenterState } from './AppPresenterState'
 
 const initalState: AppPresenterState = {
+  networkNames: [],
+  selectedNetworkName: null,
+  selectedNetworkWsUrl: null,
   navMenuVisible: false,
+  enableMainnet: import.meta.env.VITE_ENABLE_MAINNET === 'true',
 }
 
 export class AppPresenter extends Presenter<AppPresenterState> {
   private createTransactionService: CreateTransactionService
+  private switchNetworkService: SwitchNetworkService
+  private getSelectedNetworkService: GetSelectedNetworkService
+  private listNetworks: ListNetworksService
 
   constructor(
-    createTransactionService: CreateTransactionService
+    createTransactionService: CreateTransactionService,
+    switchNetworkService: SwitchNetworkService,
+    getSelectedNetworkService: GetSelectedNetworkService,
+    listNetworks: ListNetworksService,
   ) {
-    super(initalState)
+    super({
+      ...initalState,
+    })
     this.createTransactionService = createTransactionService
+    this.switchNetworkService = switchNetworkService
+    this.getSelectedNetworkService = getSelectedNetworkService
+    this.listNetworks = listNetworks
+    this.init()
+  }
+
+  private async init() {
+    await this.loadNetworkState()
+  }
+
+  private async loadNetworkState() {
+    const response = await this.listNetworks.execute()
+    const networks = response.networks
```

Comment:
Smaller one-liner that does the same 😅 

```suggestion
    const { networks } = await this.listNetworks.execute()
```

Reply:
Done. Thanks!